apps/cowswap-frontend/src/modules/trade/hooks/setupTradeState/useSetupTradeState.ts

```diff
@@ -206,7 +206,7 @@ export function useSetupTradeState(): void {
 
     if (!providerChainId || providerChainId === currentChainId) return
 
-    const targetChainId = rememberedUrlStateRef.current?.chainId || currentChainId
+    const targetChainId = urlChainId ?? rememberedUrlStateRef.current?.chainId ?? currentChainId
```

Comment:
Does the fix include also accepting `0` or `''` as `targetChainId`?
Otherwise it could remain with the `||` operator instead of `??`